src/BenchmarkDotNet/Toolchains/Roslyn/Builder.cs

```diff
@@ -11,9 +12,72 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using OurPlatform = BenchmarkDotNet.Environments.Platform;
+using MonoRuntime = BenchmarkDotNet.Environments.MonoRuntime;
 
 namespace BenchmarkDotNet.Toolchains.Roslyn
 {
+    [PublicAPI]
+    public class MonoAotBuilder : Builder
+    {
+        MonoRuntime mono;
+
+        [PublicAPI]
+        public override BuildResult Build(GenerateResult generateResult, BuildPartition buildPartition, ILogger logger)
+        {
+            BuildResult result = base.Build (generateResult, buildPartition, logger);
+
+            if (!result.IsBuildSuccess)
+                return result;
+
+            var dllPath = generateResult.ArtifactsPaths.ExecutablePath;
+            var monoPath = mono.CustomPath ?? "mono";
+
+            try {
+                using (var process = new Process { StartInfo = CreateMonoAot(monoPath, dllPath) })
+                {
+                    logger.WriteLineInfo("// Execute: " + process.StartInfo.FileName + " " + process.StartInfo.Arguments);
+                    ConsoleHandler.Instance.SetProcess(process);
+                    process.Start();
+                    process.WaitForExit(); // should we add timeout here?
+
+                    if (process.ExitCode != 0)
+                        logger.WriteLineError("Attempt to AOT failed: " + process.StartInfo.FileName + " " + process.StartInfo.Arguments);
+                }
+            }
+            finally
+            {
+                ConsoleHandler.Instance.ClearProcess();
+            }
+
+            return result;
+        }
+
+        private ProcessStartInfo CreateMonoAot(string monoPath, string filePath)
+        {
+            var absFilePath = Path.GetFullPath(filePath);
+            var start = new ProcessStartInfo
+            {
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                WorkingDirectory = Path.GetDirectoryName(absFilePath)
+            };
+            start.FileName = monoPath;
+            start.Arguments = String.Format("{0} {1}", mono.AotArgs, absFilePath);
+            if (mono.MonoBclPath != null)
+                start.EnvironmentVariables ["MONO_PATH"] = mono.MonoBclPath;
+
+            return start;
+        }
+
+        public MonoAotBuilder (MonoRuntime monoReference)
```

Comment:
please move the ctor to the top of the file, also it's a good place to use expression `=>`